### PR TITLE
prometheus-junos-czerwonk-exporter: init at 0.10.1 + module

### PR DESCRIPTION
**Pull request by NetaliDev**
###### Description of changes

This PR adds the JunOS Prometheus exporter package and a NixOS module for it.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ajs124** [comment]
How does this relate to https://github.com/NixOS/nixpkgs/pull/235423?

**Comment by NetaliDev** [comment]
It's actually not related at all, it's just a funny coincidence that we opened a PR for the same package on the same day (with this PR additionally containing a module).

But I am unsure how to proceed now, since both PRs want to add different junos_exporters as a package.

**Comment by Janik-Haag** [comment]
Rename both of them to `prometheus-juneos-exporter-${repo-owner}` ?

**Comment by NetaliDev** [comment]
Sounds good to me, but what about the module? Do you also plan to add a module for the exporter you are adding? If yes, I would do the same naming for the module.

**Comment by Janik-Haag** [comment]
Yes I planed to write a module at some point, keeping the naming consistent there sounds good.

File: pkgs/servers/monitoring/prometheus/junos-exporter.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  };
+
+  vendorSha256 = "sha256-IV0FZb1rjOMLf+vkzz/ZxUBMFD8VRDS51Wdud/yz32E=";
+
+  meta = with lib; {
+    description = "Exporter for metrics from devices running JunOS";
+    homepage = "https://github.com/czerwonk/junos_exporter";
+    license = licenses.mit;
+    maintainers = with maintainers; [ netali ];
+    platforms = platforms.unix;
```
Comment (Janik-Haag) [inline pkgs/servers/monitoring/prometheus/junos-exporter.nix:21]: Why are you actively limiting this to unix platforms?

Comment (NetaliDev) [reply pkgs/servers/monitoring/prometheus/junos-exporter.nix:21]: Whoops, thats a copy-paste remnant, I will remove this when I rename the package.

Comment (Janik-Haag) [reply pkgs/servers/monitoring/prometheus/junos-exporter.nix:21]: The rest looks good to me

Comment (NetaliDev) [reply pkgs/servers/monitoring/prometheus/junos-exporter.nix]: Fixed

**Review by Janik-Haag — APPROVED** [review]
LGTM

**Comment by NetaliDev** [comment]
Renaming is done, but I chose `prometheus-junos-${repo-owner}-exporter` as the naming, because all the other exporters end with "exporter" and I don't want to have a inconsistency there.

**Comment by Janik-Haag** [comment]
@ajs124 can you merge this?

**Pull request merged**